### PR TITLE
Feature#1121 site sync add site information and buttons to loader

### PR DESCRIPTION
**Pull request by kalisp**
Will close pypeclub/pype/issues/1121

Requires:
pypeclub/pype#1162

### CODE REVIEW

File: avalon/tools/loader/app.py [hunk 1]
```diff
@@ -249,6 +253,9 @@ def on_refreshed(has_item):
 
         self.data["state"]["assetIds"] = asset_ids
 
+        representations = self.data["widgets"]["representations"]
+        representations.model.set_version_ids([])  # reset repre list
```
Comment (iLLiCiTiT) [inline avalon/tools/loader/app.py:257]: It is better not call methods on attribute of widget. Rather implement `set_version_ids` on representations widget that wraps that.
```
def set_version_ids(self, version_ids):
    self.model.set_version_ids(version_ids)
````

This way you'll prevent touching of inner logic of representations widget in future. Like when you'll need to do something before it is called on model.

File: avalon/tools/loader/model.py [hunk 2]
```diff
@@ -109,6 +111,15 @@ def __init__(
         self.asset_doc_projection = asset_doc_projection
         self.subset_doc_projection = subset_doc_projection
 
+        self.sync_server_settings = None
+        proj_settings = get_project_settings(dbcon.Session["AVALON_PROJECT"])
+        if proj_settings.get('global').get('sync_server')['enabled']:
```
Comment (iLLiCiTiT) [inline avalon/tools/loader/model.py:116]: If you don't specify default return value of get method use rather `proj_settings['global']['sync_server']['enabled']`. Result is the same but shorted.

File: avalon/tools/loader/model.py [hunk 3]
```diff
@@ -109,6 +111,15 @@ def __init__(
...
+            self.sync_server_settings = proj_settings.get('global').\
+                get('sync_server')
+            self.Columns.append("repre_info")
+            self.column_labels_mapping["repre_info"] = "Repre info"
+            self.repre_icons = get_repre_icons()
```
Comment (iLLiCiTiT) [inline avalon/tools/loader/model.py:121]: Attribute `self.repre_icons` is not specified if syns settings are not enabled.

File: avalon/tools/loader/model.py [hunk 4]
```diff
@@ -726,3 +815,142 @@ def sort(self, column, order):
...
+
+        self.endResetModel()
+        self.refreshed.emit(False)
+
+    def refresh(self):
+
+        if self.version_ids:
+            # Simple find here for now, expected to receive lower number of
+            # representations and logic could be in Python
+            self._docs = self.dbcon.find(
```
Comment (iLLiCiTiT) [inline avalon/tools/loader/model.py:896]: Result of `find` is iterable, should be converted to list to keep the data.

File: avalon/tools/loader/model.py [hunk 5]
```diff
@@ -726,3 +815,142 @@ def sort(self, column, order):
...
+                {"type": "representation", "parent": {"$in": self.version_ids}}
+                , self._get_projection()
+            )
+
+        self.doc_fetched.emit()
+
+    def _get_projection(self):
```
Comment (iLLiCiTiT) [inline avalon/tools/loader/model.py:903]: Rather use class attribute for projections then method. This way the same dictionary is created again and again.

**Review by mkolar — APPROVED** [review]

**Pull request merged**